components/formatted_markdown_message.jsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {injectIntl, intlShape} from 'react-intl';
+import PropTypes from 'prop-types';
+import marked from 'marked';
+
+const TARGET_BLANK_URL_PREFIX = '!';
+
+class CustomRenderer extends marked.Renderer {
+    link(href, title, text) {
+        if (href[0] === TARGET_BLANK_URL_PREFIX) {
+            return `<a href="${href.substring(1, href.length)}" target="_blank">${text}</a>`;
+        }
+        return `<a href="${href}">${text}</a>`;
+    }
+
+    paragraph(text) {
+        return text;
+    }
```

Comment:
@hmhealey I made a change to the already-reviewed `<FormattedMarkdownMessage>` component (lines 19-21 added):

It now overrides the `paragraph` method of the default renderer to remove the surrounding `<p>` tags that markedjs adds to all markdown strings. 

The reason for this is that `<FormattedHTMLMessage>` doesn't add surrounding `<p>` tags, so in order for `<FormattedMarkdownMessage>` to be a drop-in replacement the `<p>` tags cannot be used or else styling changes may be required for all instances of `<FormattedMarkdownMessage>`.

If we ever want to use `<FormattedMarkdownMessage>` for block level markdown (like blockquotes) then more methods may need to overridden, but since the plan is to just use `<FormattedMarkdownMessage>` for inline markdown it should be OK.

Let me know if you see any concerns with this approach.

Reply:
That sounds good to me. I figure this will most often be used for single lines of text anyway, so the paragraph tags aren't necessary. If we decide we want them somewhere, we could even add a prop to the component that turns them on or off as needed.